api/src/mutations/vault.rs

```diff
@@ -62,7 +62,7 @@ impl Mutation {
         let (customer_treasury, treasury) = customer_treasuries::Entity::find()
             .join(
                 JoinType::InnerJoin,
-                customer_treasuries::Relation::Treasury.def(),
+                customer_treasuries::Relation::Treasuries.def(),
```

Comment:
Quality of life but this seems like it should be singular since customer can only have 1 treasury.